base/deprecated.jl

```diff
@@ -826,6 +826,7 @@ end
 # Deprecate vectorized !
 @deprecate(!(A::AbstractArray{Bool}), .!A) # parens for #20541
 @deprecate(!(B::BitArray), .!B) # parens for #20541
+!(::typeof(!)) = identity # make sure ! has at least 4 methods so that for-loops don't end up getting a back-edge to depwarn
```

Comment:
I think it's a bit of a gotcha if `!!` accepts arguments that `!` doesn't. Also, not too long from now the two deprecated methods of `!` will be removed, which also removes the need for this method (IIUC), but in the meantime code could have become dependent on this behavior. So I think it would be helpful to type-restrict this.